artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/TypedProperties.java

```diff
@@ -318,6 +313,30 @@ public synchronized boolean containsProperty(final SimpleString key) {
       }
    }
 
+   public synchronized boolean removeProperty(Predicate<SimpleString> propertyNamePredicate) {
```

Comment:
I am a bit confused. you said you're optimizing, although this is a new method. what I am missing?

Reply:
Yep, then original version was using a slower method to remove a property matching specific criteria and it was used in the hot path for every message.
This change has improved (for small binary message) the throughput of core on my tests by a perceivable quantity.